esphome/components/cm1106/sensor.py

```diff
@@ -0,0 +1,73 @@
+"""CM1106 Sensor component for ESPHome."""
+
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome import automation
+from esphome.automation import maybe_simple_id
+from esphome.components import sensor, uart
+from esphome.const import (
+    CONF_CO2,
+    CONF_ID,
+    DEVICE_CLASS_CARBON_DIOXIDE,
+    ICON_MOLECULE_CO2,
+    STATE_CLASS_MEASUREMENT,
+    UNIT_PARTS_PER_MILLION,
+)
+
+DEPENDENCIES = ["uart"]
+CODEOWNERS = ["@andrewjswan"]
+
+cm1106_ns = cg.esphome_ns.namespace("cm1106")
+CM1106Component = cm1106_ns.class_(
+    "CM1106Component", cg.PollingComponent, uart.UARTDevice
+)
+CM1106CalibrateZeroAction = cm1106_ns.class_(
+    "CM1106CalibrateZeroAction",
+    automation.Action,
+)
+CM1106ABCEnableAction = cm1106_ns.class_("CM1106ABCEnableAction", automation.Action)
+CM1106ABCDisableAction = cm1106_ns.class_("CM1106ABCDisableAction", automation.Action)
+
+CONFIG_SCHEMA = (
+    cv.Schema(
+        {
+            cv.GenerateID(): cv.declare_id(CM1106Component),
+            cv.Required(CONF_CO2): sensor.sensor_schema(
```

Comment:
No sensors are required in ESPHome. If this is the only sensor for this component, then the entire `CONFIG_SCHEMA` should be an extension of `sensor.sensor_schema`.

Example:
https://github.com/esphome/esphome/blob/6f801fec66b7b2b5e4f7749afa14eaf271abebb4/esphome/components/a01nyub/sensor.py#L18-L26

```suggestion
            cv.Optional(CONF_CO2): sensor.sensor_schema(
```

Reply:
I see there are some Required sensors in ESPHome, this is wrong and I am going to change them. Please alter this to be Optional.

I am going to also update https://developers.esphome.io with this requirement. Currently it has only been specified adhoc in PR reviews when it came up (and we didnt have developer docs)